core/authority-discovery/primitives/src/lib.rs

```diff
@@ -20,31 +20,28 @@
 
 use client::decl_runtime_apis;
 use rstd::vec::Vec;
-use sr_primitives::RuntimeDebug;
 
-#[derive(codec::Encode, codec::Decode, Eq, PartialEq, Clone, RuntimeDebug)]
-#[cfg_attr(feature = "std", derive(Hash))]
-pub struct Signature(pub Vec<u8>);
-#[derive(codec::Encode, codec::Decode, Eq, PartialEq, Clone, RuntimeDebug)]
-#[cfg_attr(feature = "std", derive(Hash))]
-pub struct AuthorityId(pub Vec<u8>);
+mod app {
+	use app_crypto::{app_crypto, key_types::AUTHORITY_DISCOVERY, sr25519};
```

Comment:
Hmm, I think we should follow `im-online` and provide support for `sr25519` and `ed25519`. The module probably does not have a hard requirement on any crypto.

Reply:
Right now I am using the `BareCryptoStore` trait to access the keystore within the authority discovery module. The trait only offers specific functions for sr25519 and ed25519, which I could not use if we make the crypto type generic. Are you suggesting to access the `Store` directly and use `key_pair` to retrieve the key pair?

Reply:
i don't think it's necessary at this stage, and it wouldn't be the only module that supports a specific type of crypto.